api/v1alpha1/ratelimit_types.go

```diff
@@ -51,6 +51,16 @@ type GlobalRateLimit struct {
 	//
 	// +kubebuilder:validation:MaxItems=64
 	Rules []RateLimitRule `json:"rules"`
+
+	// Shared determines whether the rate limit rules apply across all the policy targets.
+	// If set to true, the rule is treated as a common bucket and is shared across all policy targets (xRoutes).
+	// Must have targetRef set to Gateway
+	// Default: false.
+	//
+	// +optional
+	// +notImplementedHide
+	// +kubebuilder:default=false
+	Shared *bool `json:"shared,omitempty"`
```

Comment:
```suggestion
	BucketScope RateLimitBucketScope `json:"bucketScope,omitempty"`
```
You can define enums:
```go
type RateLimitBucketScope string

const (
	SharedBucket    RateLimitBucketScope = "shared"    // Shared across routes
	PerRouteBucket  RateLimitBucketScope = "perRoute" // Independent for each route, default
)
```

Reply:
+1 to future proof this API for other scopes (e.g. vhost?)

Reply:
prefer just `scope`
and instead of `perRoute` it may help to define it as `perTarget`

Reply:
Just scope might be confusing with the rate limit scope, i.e, Global/Local. But I guess that could be clarified in the documentation.

Reply:
I could see the confusion with `scope` being confused with `type` (Global/Local), `bucketScope` could provide more clarity. Also agree with  @arkodg  for `perTarget` rather than `perRoute`.

I'll push through whichever changes we agree upon here

Reply:
i was trying to avoid  the `bucket` prefix, because i was unsure if it should be `bucket`, `rule` (which is an existing noun in the API) or `somethingElse`
we can brainstorm a little more here

Reply:
How about "context" (values are shared, perTarget, perVHost, etc)? I am ok with "scope" as well since it's explicitly defined under GlobalRateLimit API, just that context feels less confusing.

Reply:
though i see the potential confusion, `scope: shared/target/vHost` does seem most clear

can the `vHost` option be added in another patch? was hoping to just focus on shared/target

Reply:
Yeah we can't add vHost yet, because that noun doesn't exist in any user facing docs

Reply:
lets try and unblock Ryan, sharing some options, please vote
@envoyproxy/gateway-maintainers @envoyproxy/gateway-reviewers 
a) `bucketScope: Gateway | PerTarget`
b) `scope: Gateway | PerTarget`
c) `context: Gateway | PerTarget`

Reply:
a

Reply:
+1 for Option b

Reply:
+1 for b

I've been developing the feature just using shared=true field for now, can readjust once we have a decision - so i'm not blocked

Reply:
+1 for b